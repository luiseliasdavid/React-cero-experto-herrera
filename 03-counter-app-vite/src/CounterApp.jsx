import React, {useState} from 'react';
import PropTypes from "prop-types"

export const CounterApp = ({value}) => {

   const [counter, setCounter] = useState(value)

   const handleClick= (e)=>{
    console.log(e)
    //setCounter(counter + 1)
    
    e=="+1" ? setCounter(counter + 1) :
    e=="-1" ? setCounter(counter - 1): 
     setCounter(value);
    
   }
   


    return (
        <>
            <h1  >CounterApp</h1>
            <h2 data-testid="h2"> {counter} </h2>
            <button  data-testid="+1" onClick={()=>handleClick("+1")}>+1</button>
            <button onClick={()=>handleClick("-1")} >-1</button>
            <button  aria-label='btn-reset' onClick={()=>handleClick("reset")}>Reset</button>
        </>
    );
}

CounterApp.propTypes = {
value : PropTypes.number.isRequired,

}


