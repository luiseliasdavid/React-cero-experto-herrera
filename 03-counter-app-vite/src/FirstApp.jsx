import PropTypes from "prop-types"

export const FirstApp = ({title, subtitle, name}) => {

    
    return (
        <div>
            <h1>{title} </h1>
            <h3> {subtitle} </h3>
            <h4> {name} </h4>
        </div>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    name: PropTypes.string
}

//FirstApp.defaultProps
