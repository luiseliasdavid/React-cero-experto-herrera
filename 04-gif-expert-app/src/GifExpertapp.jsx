import { useState } from "react"
import { AddCategory } from "./components/AddCategory"

export const GifExpertApp = () => {

const [category,setCategory]= useState(['element1', 'element2'])

const agregarElement = (element)=> {
  if(category.includes(element)) return;
  setCategory([...category,element])
}

  return (
   <>
    <h1>GifExpertApp</h1>

    <AddCategory agregarElement={agregarElement} />
    
    <ol>
      {category.map(element =>{return  <li key={element} >{ element} </li>})}
    </ol>
   </>
  )
}


