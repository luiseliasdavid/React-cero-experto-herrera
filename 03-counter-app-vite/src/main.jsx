import React from "react"
import ReactDOM from "react-dom/client"
import { CounterApp } from "./CounterApp"
import { FirstApp } from "./FirstApp"
import {App} from "./HelloWorldApp"
import "./main.css"



ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
       {/*  <App /> */}
       <FirstApp title="hola soy luis" subtitle='no hay subtitulo' name='luis david'/>
       {/* <CounterApp value={20} /> */}
    </React.StrictMode>
)