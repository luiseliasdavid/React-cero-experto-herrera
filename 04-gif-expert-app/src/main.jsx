import React from 'react';
import ReactDOM from 'react-dom/client';
import {GifExpertApp} from './GifExpertapp';
import './styles.css'
//99MP4tSHpTcUx3RrsAdKvJIQHoIjtTV7

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>
)
