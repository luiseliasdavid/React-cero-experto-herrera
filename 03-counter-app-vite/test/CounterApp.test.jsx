import { getByTestId, render, screen } from '@testing-library/react'
import { CounterApp } from '../src/CounterApp'

describe('Probando counter app', () => {
  const initialValue = 100

  test('should  match with snapshot', () => {
    
    const {container} = render(<CounterApp value={initialValue} />)

    expect(container).toMatchSnapshot(); 
  })
  test('counter should be 100 ', () => {
      render(<CounterApp value={initialValue} />)
      screen.debug()
    // expect(screen.getByText(100)).toByTruthy();
    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(String(initialValue))
  })
  


})


