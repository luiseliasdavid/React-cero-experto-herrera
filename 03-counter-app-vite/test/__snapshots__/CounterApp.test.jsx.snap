// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando counter app should  match with snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="h2"
  >
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
